### ISSUE TITLE
Module Java Refactoring threw java.lang.IllegalStateException

### ISSUE THREAD

**Issue by lostbits** [apache/netbeans#7005]
### Apache NetBeans version

Apache NetBeans 20

### What happened

Attempted for refactor name of a class. The message.log is included.

The message received was:
`Module Java Refactoring threw java.lang.IllegalStateException: java.lang.NullPointerException: Cannot invoke "com.sun.tools.javac.code.Symbol$ClassSymbol.isDirectlyOrIndirectlyLocal()" because "tree.sym" is null. Please report a bug against Java Refactoring module and attach your var/log/messages.log.`

[messages.log](https://github.com/apache/netbeans/files/14068894/messages.log)

### How to reproduce

I don't know if it is reproducible. What I did was to highlight the class name and then refactor->rename.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 11 Pro v23H2 Experience Pack 1000.22681.1000.0

### JDK

jdk-17.0.1

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

Just tried it (once). Will not retry until I hear that the issue is resolved.

### Are you willing to submit a pull request?

No

**Comment by neilcsmith-net** [comment]
JDK 17.0.1 is old.  Please try with a current release of JDK 17 or 21.

Is the `slipCell` class mentioned in the log correctly coded?  Please attach the dump files mentioned to this issue.

Lot of `is set to: 17, java.util.zip.CRC32C cannot be found on the system module path:` messages in the log.

cc/ @lahodaj @mbien any thoughts on this?

> Just tried it (once). Will not retry until I hear that the issue is resolved.

This is not a helpful statement to make!  If we don't get enough input and testing to understand what is going wrong for you, then the issue won't be resolved.

**Comment by lostbits** [comment]
i Neil;

Thanks for telling me I had a really old jdk. Replaced it with JDK-21 with the following result. Refactor->rename worked up to changing the class name (slipCell to SlipCell) without error. But this change was nowhere else made. For example, in the class there is a function:
    `  slipCell method() {}`

and although the class name was changed, the return type in the method description did not.

And so in a fit of pique I used some Linux commands and changed them all by hand. Ah, the life of a programmer is beset with difficulty.

Thanks for the time you spent. Sorry that I didn't get back to you sooner. I 'lost' the email somewhere and only after a desultory search found it.

I am really sorry if I've cause any anguish. I did not mean to do that.

art
